.github/workflows/e2e_test.yaml

```diff
@@ -80,6 +83,28 @@ jobs:
         run: |
           echo "docs: ${{ steps.e2e-test-create.outputs.index_url }}" >> metadata.yaml
           cat metadata.yaml
+      - name: Prepare for update test
+        id: e2e-test-prepare-update
+        run: |
+          # Update the contents of one of the files, do it first to get around content conflicts
+          echo -n more content >> docs/doc.md
+          GITHUB_TOKEN='${{ secrets.GITHUB_TOKEN }}'
+          REPO='${{ github.repository }}'
+          PYTHONPATH=$(pwd) python3 prepare_check_cleanup/reconcile.py --action prepare-update --action-kwargs "{\"github_token\": \"$GITHUB_TOKEN\", \"repo\": \"$REPO\", \"filename\": \"docs/doc.md\"}" '${{ steps.e2e-test-create.outputs.urls_with_actions }}' '${{ steps.configuration.outputs.discourse }}'
```

Comment:
What would be the benefit? Yes this could be improved in a bunch of ways, for example, a more general improvement would be to use something like `click`. Since this is mainly used for e2e test setup and teardown and not exposed to end users it investing in it doesn't seem like it would benefit a lot. At best, it could improve the readability of the e2e test workflow

Reply:
I was thinking in terms of reducing maintenance cost by having better readability and structured argument parsing, which gives better visibility within the module on the data that is needed. But after seeing the reasoning for kwargs, i am split between readability issue in json parsing vs having more boilerplate code. Thank you for the reply!